modules/unsupported/mbtiles/src/main/java/org/geotools/mbtiles/ClippingFeatureReader.java

```diff
@@ -0,0 +1,87 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2020, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.mbtiles;
+
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import org.geotools.data.simple.SimpleFeatureReader;
+import org.geotools.util.factory.Hints;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.Polygon;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.GeometryDescriptor;
+
+/**
+ * The class takes care to return features that intersects the geometry passed with a
+ * Hints.GEOMETRY_CLIP, in order to avoid that features' buffer in mbtiles data gets deployed when a
+ * rendering transformation is issued.
+ */
+class ClippingFeatureReader implements SimpleFeatureReader {
```

Comment:
This could be easily turned into a DecoratingFeatureIterator instead, which a ClippingFeatureCollection would use.

Reply:
fixed